Packs/CommonScripts/Scripts/ReadPDFFileV2/ReadPDFFileV2.py

```diff
@@ -260,16 +251,117 @@ def get_urls_and_emails_from_pdf_html_content(cpy_file_path, output_folder):
     Args:
         cpy_file_path (str): the path of the PDF file.
         output_folder (str): the folder output to get the HTML files from.
-
     Returns:
         tuple[set, set]: The URLs and emails that were found.
     """
     pdf_html_content = get_pdf_htmls_content(cpy_file_path, output_folder)
     return set(re.findall(URL_EXTRACTION_REGEX, pdf_html_content)), set(re.findall(EMAIL_REGXEX, pdf_html_content))
 
 
+def decrypt_pdf_file(file_path, user_password, path_to_decrypted_file):
+    """
+    Gets a path to an encrypted PDF file and its password, and decrypts the file using pikepdf package.
+    Args:
+        file_path (str): A path to the encrypted PDF file.
+        user_password (str): The password to the encrypted PDF file.
+        path_to_decrypted_file (str): A path to save the decrypted PDF file to.
+
+    Returns: None.
+
+    """
+    pdf_file = Pdf.open(file_path, password=user_password)
+    pdf_file.save(path_to_decrypted_file)
+
+
+def get_urls_and_emails_from_pdf_annots(file_path):
+    """
+    Extracts the URLs and Emails from the pdf's Annots (Annotations and Commenting) using PyPDF2 package.
+    Args:
+        file_path (str): The path of the PDF file.
+
+    Returns: A set includes the URLs that were found, A set includes the Emails that were found
+
+    """
+    urls_set = set()
+    emails_set = set()
+
+    pdf_file = open(file_path, 'rb')
+    pdf = PyPDF2.PdfFileReader(pdf_file)
+    pages = pdf.getNumPages()
+
+    # Goes over the PDF, page by page, and extracts urls and emails:
+    for page in range(pages):
+        page_sliced = pdf.getPage(page)
+        page_object = page_sliced.getObject()
+
+        # Extracts the PDF's Annots (Annotations and Commenting):
+        if annots := page_object.get('/Annots'):
+            if not isinstance(annots, PyPDF2.generic.ArrayObject):
+                annots = [annots]
+
+            for annot in annots:
```

Comment:
I understand what you're saying ... but I have a problem splitting it into 3 functions because I do not want to pass by reference the urls_set and the emails_set to each of the functions.

Instead, I created a separate function for extracting the URL from the annot_object and added more comments to the main function. 
Please, let me know what you are thinking.